### PR TITLE
Problem: duplicate lib.rs.cc symbols when used with defi-wallet-core

### LINKED ISSUE #89: Problem: duplicate lib.rs.cc symbols when used with defi-wallet-core
some symbols are duplicate,
if two frameworks are all shared-lib, it's OK
but when used with static-lib, cannot build into one binary

```
Invalidating makefile for DefiWalletCoreUnrealEditor (source file added)
ERROR: Unable to merge actions producing lib.rs.cc.o: prerequisites are different
ERROR: Unable to merge actions producing lib.rs.cc.o: command arguments are different
ERROR: Unable to merge actions producing lib.rs.cc.o: prerequisites are different
ERROR: Unable to merge actions producing lib.rs.cc.o: command arguments are different
ERROR: Action graph is invalid; unable to continue. See log for additional details
```

1. find a way to change cxxbridge1 namespace
or
2. expose together in one build.rs
- maybe play-cpp-sdk can contain all as one crate

### LINKED ISSUE #94: Problem: Duplicate name lib.rs.cc in windows CI build
```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(1091,5): warning MSB8027: Two or more files with the name of lib.rs.cc will produce outputs to the same location. This can lead to an incorrect build result.  The files involved are include\extra-cpp-bindings\src\lib.rs.cc, include\defi-wallet-core-cpp\src\lib.rs.cc. [D:\a\play-cpp-sdk\play-cpp-sdk\demo\demo.vcxproj]
```

### PR DESCRIPTION
**Pull request by elecming**
Close:
- #89
- #94

Solution:
- Add defi-wallet-core-rs as submodule
- Add wrapper crate play-cpp-sdk

### CODE REVIEW

**Comment by leejw51crypto** [comment]
git submodule errors?
```
repository 'https://github.com/crypto-com/defi-wallet-core-rs.git/' not found
```

**Comment by elecming** [comment]
@leejw51crypto yes, requesting a machine user.

**Comment by elecming** [comment]
only windows CI problem left.

**Review by tomtau — COMMENTED** [review]
can the windows build be fixed?

**Comment by elecming** [comment]
@tomtau Yup, I will fix it today.

**Comment by elecming** [comment]
Don't know why it failed in windows, I am setting up in windows 10 and try... @leejw51crypto

**Comment by leejw51crypto** [comment]
i'm checking too, seems some files are not linked

**Comment by leejw51crypto** [comment]
reproduced the same error,

https://docs.microsoft.com/en-us/answers/questions/423205/lnk4042-object-specified-more-than-once-extras-ign.html

it's by the visual studio itself,
mac, linux: clang
windows: visual studio , duplicate symbol not allowed

**Comment by leejw51crypto** [comment]
play_cpp_sdk.lib on windows works (static lib, about 120 MB)
only dll.lib on windows not working

**Comment by elecming** [comment]
One remaining Error: ``` LINK : fatal error LNK1181: cannot open input file 'lib\play_cpp_sdk.lib```, I am checking.

**Comment by elecming** [comment]
Ready for review~ @tomtau @leejw51crypto.
For windows, only staticlib is supported at this moment. https://github.com/crypto-com/play-cpp-sdk/issues/95

And one warning because of duplicate name `lib.rs.cc`, https://github.com/crypto-com/play-cpp-sdk/issues/94.

Other warnings are from `easywclient.cpp`, likely could be ignored at this moment or change other websocket library in the future.

**Comment by elecming** [comment]
Fixed #94 by moving `lib.rs.cc` as `core.cc`

**Comment by leejw51crypto** [comment]
lgtm,

some comments would be helpful to users.(markdown or Makefile?)

1. about `x64 native command prompt` for building cargo
2. and remove ~/.cargo for full rebuild for git residue
3. for windows, play-cpp-sdk.dll.lib is not supported, and static lib is supported for now

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          token: ${{ secrets.PAT }}
+          submodules: recursive
+      - uses: cachix/install-nix-action@v15
+      - uses: cachix/cachix-action@v10
+        with:
+          name: cronos
+          extraPullNames: dapp
+          # github don't pass secrets for pull request from fork repos,
+          # in that case the push is disabled naturally.
+          signingKey: "${{ secrets.CACHIX_SIGNING_KEY }}"
```
Comment (tomtau) [inline .github/workflows/docs.yml]: why does it need "cronos" and a signing key?
I assume it's not updating a cache?

**Comment by elecming** [comment]
OK, good to add~

**Review by tomtau — COMMENTED** [review]
likely ok... Maybe the nix build could use nix flakes instead.

anyway, one thing I'm not sure is why the gh action needs cachix signing key?

File: .github/workflows/docs.yml (hunk 1 above)
Comment (elecming) [reply .github/workflows/docs.yml]: ok, l'll remove it..

Comment (elecming) [reply .github/workflows/docs.yml]: Removed all nix and documents related codes. Will be done in another PR.

**Comment by elecming** [comment]
Ready again~

**Pull request merged**